app/models/game.rb

```diff
@@ -21,4 +22,18 @@ def populate_board!
       count += 1
     end
   end
+
+  def white_king_check?
+    pieces.where(color: 'black').each do |piece|
+      return true if piece.valid_move?(@white_king.horizontal_position, @white_king.vertical_position)
+      false
```

Comment:
I think the `false` will work here since we're not returning it until we get to the last iteration. But I think it would just be clearer to put it after the each loop finishes.